src/parser/replace-values-parser.ts

```diff
@@ -0,0 +1,40 @@
+import { ColumnNode } from '../operation-node/column-node.js'
+import { ValuesNode } from '../operation-node/values-node.js'
+import {
+  InsertType,
+  NonNullableInsertKeys,
+  NonNullableUpdateKeys,
+  NullableInsertKeys,
+  NullableUpdateKeys,
+  UpdateType,
+} from '../util/column-type.js'
+import { parseInsertObjectOrList } from './insert-values-parser.js'
+import { ValueExpression } from './value-parser.js'
+
+export type ReplaceObject<DB, TB extends keyof DB> = {
```

Comment:
Why is this type needed? Isn't the input for `replace into` exactly the same as for `insert into`? The only difference is that mysql deletes the old row if it exists. I should not affect the types.

Reply:
To my understanding, the argument of `replaceInto().values(arg)` differs from the argument of `insertInto().values(arg)` in that it should accept `ColumnType<any, never, any>` columns, while the latter should not. `replaceInto` is the permissive "everything goes" brother of `insertInto`, getting the values exactly where you expect them to be at all costs.

example:

```typescript
type Person {
  id: Generated<number> // auto-increment
  first_name: string
  middle_name: string | null
  last_name: string
  created_at: GeneratedAlways<string>
}

type Database {
  persons: Person
}

const person = {
  id: 15,
  first_name: 'Jennifer',
  last_name: 'Aniston',
}

db<Database>.insertInto('persons').values([person]) // typescript error at `[person]` (due to Generated<number>), which is fine, we want to insert new person and let the database generate id for us.

db<Database>.replaceInto('persons').values([person]) // should be OK! we want to insert a person we got, and if exists with id:15, to replace it.
```

Reply:
Not a big deal though, I'm OK with not having this difference between the two.

Reply:
Actually this is ok

```ts
const person = {
  id: 15,
  first_name: 'Jennifer',
  last_name: 'Aniston',
}

db.insertInto('persons').values([person]) 
```

You can insert values to auto incrementing rows. There's no typescript error in that case.

Reply:
You're right, my bad. 💯